codec_fixed.go

```diff
@@ -140,7 +146,20 @@ func (c *fixedDecimalCodec) Decode(ptr unsafe.Pointer, r *Reader) {
 }
 
 func (c *fixedDecimalCodec) Encode(ptr unsafe.Pointer, w *Writer) {
-	r := *((**big.Rat)(ptr))
+	var r *big.Rat
+	switch c.typ.Kind() {
+	case reflect.Ptr:
+		r = *((**big.Rat)(ptr))
+	case reflect.String:
+		s := *((*string)(ptr))
+		var ok bool
+		r, ok = (&big.Rat{}).SetString(s)
```

Comment:
Same issue here, it will sometimes work and sometimes not. I would argue that data sanitisation is not the role of this package.

Reply:
In this case we can encode to 0 if `.SetString()` returns not ok, instead of returning an error.  I had the same concern here as well about silently encoding a different value.